### PR TITLE
Fix channel issues

### LINKED ISSUE #54: BUG: HandleJoin crashes when a user comes back
When a user QUITs and joins back, the following line crashes
https://github.com/SirCmpwn/ChatSharp/blob/master/ChatSharp/Handlers/ChannelHandlers.cs#L15

Attached is the log of the bot. A good solution would be to remove the user entirely from the user collection once he/she quits.
[chatsharp_bot_log.txt](https://github.com/SirCmpwn/ChatSharp/files/131337/chatsharp_bot_log.txt)

### PR DESCRIPTION
**Pull request by RockyTV**
Fixes #54.

This is a series of commits trying to fix the issue described in #54. The actual fix was pretty simple: just mirror `client.Channels` to `client.User.Channels`. With that done, there is no need for a special logic to check if the current user who parted the channel is the library client and act accordingly.

However, as stated in the issue linked above, I've also managed to fix the user modes bug that appeared by adding a equality check to `IrcChannel`. C#, by default, will compare two objects based on their members if the equality operator isn't overridden.

### CODE REVIEW

File: ChatSharp/Handlers/MessageHandlers.cs [hunk 1]
```diff
@@ -182,15 +182,16 @@ public static void HandleMode(IrcClient client, IrcMessage message)
                                     new UserPoolView(channel.Users.Where(u =>
                                     {
                                         if (!u.ChannelModes.ContainsKey(channel))
-                                            u.ChannelModes.Add(channel, null);
-                                        return u.ChannelModes[channel] == c;
+                                            u.ChannelModes.Add(channel, new System.Collections.Generic.List<char?>());
```
Comment (SirCmpwn) [inline ChatSharp/Handlers/MessageHandlers.cs]: Add a using statement rather than fully qualifying this.

**Comment by SirCmpwn** [comment]
LGTM aside from minor nitpicking

**Comment by RockyTV** [comment]
Any other thing you want changed?

**Comment by SirCmpwn** [comment]
Nope, thanks!

**Pull request merged**